src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2041,16 +2052,18 @@ public void close()
         private DataLimits.Counter counter;
         private PartitionIterator sentQueryIterator;
 
+        private int maxConcurrencyFactor;
         private int concurrencyFactor;
         // The two following "metric" are maintained to improve the concurrencyFactor
         // when it was not good enough initially.
         private int liveReturned;
         private int rangesQueried;
 
-        public RangeCommandIterator(RangeIterator ranges, PartitionRangeReadCommand command, int concurrencyFactor, Keyspace keyspace, ConsistencyLevel consistency, long queryStartNanoTime)
+        public RangeCommandIterator(RangeIterator ranges, PartitionRangeReadCommand command, int concurrencyFactor, int maxConcurrencyFactor, Keyspace keyspace, ConsistencyLevel consistency, long queryStartNanoTime)
```

Comment:
Nit: we could break this line
```suggestion
        public RangeCommandIterator(RangeIterator ranges,
                                    PartitionRangeReadCommand command,
                                    int concurrencyFactor,
                                    int maxConcurrencyFactor,
                                    Keyspace keyspace,
                                    ConsistencyLevel consistency,
                                    long queryStartNanoTime)
```

Reply:
+1